packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -246,10 +237,19 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20 {
                 quoteTimestamp: quoteTimestamp
             });
         bytes32 depositHash = _getDepositHash(depositData);
+
+        // If relay exists for deposit, check if it is disputed. If its disputed, then we can relay again, otherwise
+        // the relay is pending valid and we cannot re-relay.
         require(
-            (relays[depositHash].relayState == RelayState.Uninitialized ||
-                relays[depositHash].relayState == RelayState.Disputed),
-            "Deposit already relayed"
+            relays[depositHash].relayState == RelayState.Uninitialized ||
+                _getOptimisticOracle().getState(
```

Comment:
how would it be posible for the relay state to be not `Uninitialized` AND the OO having a disputed Price? put another way, why do you need to check the OO here?

Reply:
1) call `relayDeposit` for a deposit to make the relay state `Pending`.
2) Someone disputes the relay directly against the OO
3) Relay state is still `Pending` but there is an OO dispute

Thoughts?

Reply:
> The only way the OO could have invalid for this exact request is if the relay has never been relayed before, right?

Correct. But imagine that a relay is `Pending` and is then disputed directly through the OO. The `RelayState` is still `Pending`, but we should now be able to call `relayDeposit` again on it.

Reply:
Note that the above is possible and creates two separate price requests because the `relayId` nonce is incremented between relays.

Reply:
Seems reasonable to me. I also see your logic @nicholaspai.